configs/maskformer/maskformer_r50_512x512_80k_ade20k.py

```diff
@@ -0,0 +1,135 @@
+_base_ = [
+    '../_base_/datasets/ade20k.py', '../_base_/default_runtime.py',
+    '../_base_/schedules/schedule_80k.py'
+]
+norm_cfg = dict(type='SyncBN', requires_grad=True)
+crop_size = (512, 512)
+data_preprocessor = dict(
+    type='SegDataPreProcessor',
+    size=crop_size,
+    mean=[123.675, 116.28, 103.53],
+    std=[58.395, 57.12, 57.375],
+    bgr_to_rgb=True,
+    pad_val=0,
+    seg_pad_val=255)
+# model_cfg
+num_classes = 150
+model = dict(
+    type='EncoderDecoder',
+    data_preprocessor=data_preprocessor,
+    backbone=dict(
+        type='ResNet',
+        depth=50,
+        num_stages=4,
+        out_indices=(0, 1, 2, 3),
+        dilations=(1, 1, 1, 1),
+        strides=(1, 2, 2, 2),
+        norm_cfg=norm_cfg,
+        norm_eval=True,
+        style='pytorch',
+        contract_dilation=True,
+        init_cfg=dict(
+            type='Pretrained', checkpoint='open-mmlab://resnet50_v1c')),
+    decode_head=dict(
+        type='mmseg.MaskFormerHead',
+        _scope_='mmdet',
+        in_channels=[256, 512, 1024, 2048],  # pass to pixel_decoder inside
+        feat_channels=256,
+        in_index=[0, 1, 2, 3],
+        num_classes=150,
+        out_channels=256,
+        num_queries=100,
+        pixel_decoder=dict(
+            type='PixelDecoder',
+            norm_cfg=dict(type='GN', num_groups=32),
+            act_cfg=dict(type='ReLU')),
+        enforce_decoder_input_project=False,
+        positional_encoding=dict(
+            type='SinePositionalEncoding', num_feats=128, normalize=True),
+        transformer_decoder=dict(
+            type='DetrTransformerDecoder',
+            return_intermediate=True,
+            num_layers=6,
+            transformerlayers=dict(
+                type='DetrTransformerDecoderLayer',
+                attn_cfgs=dict(
+                    type='MultiheadAttention',
+                    embed_dims=256,
+                    num_heads=8,
+                    attn_drop=0.1,
+                    proj_drop=0.1,
+                    dropout_layer=None,
+                    batch_first=False),
+                ffn_cfgs=dict(
+                    embed_dims=256,
+                    feedforward_channels=2048,
+                    num_fcs=2,
+                    act_cfg=dict(type='ReLU', inplace=True),
+                    ffn_drop=0.1,
+                    dropout_layer=None,
+                    add_identity=True),
+                # the following parameter was not used,
+                # just make current api happy
+                feedforward_channels=2048,
+                operation_order=('self_attn', 'norm', 'cross_attn', 'norm',
+                                 'ffn', 'norm')),
+            init_cfg=None),
```

Comment:
Hello Mr. Li, my name is Qingyun and I participated in refactoring the DETR-like algorithm of mmdet. 
We have deconstructed DetrTransformerDecoder in the new algorithm to build with registry mechanism in open-mmlab/mmdetection#8763 (The PR has been merged into `refactor-detr` branch or official mmdetection, and will be merge to 3.x when other DETRs are refactored). 
I would like to remind you that the logic of building decoder and the config may need to be modified.

Reply:
> Hello Mr. Li, my name is Qingyun and I participated in refactoring the DETR-like algorithm of mmdet. We have deconstructed the registry mechanism to build DetrTransformerDecoder in the new implementation in [open-mmlab/mmdetection#8763](https://github.com/open-mmlab/mmdetection/pull/8763) (The PR has been merged into `refactor-detr` branch of official mmdetection repo, and will be merge to 3.x when other DETRs are refactored). I would like to remind you that the logic of building decoder and the config may need to be modified.

Thank you qingyun, very useful information for us. ;)